### ISSUE TITLE
TypeError: list_sessions() takes 1 positional argument but 2 were given

### ISSUE THREAD

**Issue by chiawchen** [jupyter-server/synchronizer#41]
## Description

Hi, thanks for implementing this fantastic client side resilient features, this is very helpful to recovery state from a client side restart, I tried it locally with the example list on readme (1 local kernelgateway, 1 local jupyterlab client), but saw below error when trying to run a regular jupyter lab with synchronizer enabled
```
.venv/lib/python3.8/site-packages/jupyter_server_synchronizer/manager.py", line 234, in list_sessions
        out = await super().list_sessions(self)
    TypeError: list_sessions() takes 1 positional argument but 2 were given
```

## Reproduce

Following the readme
1. on one terminal run the kernelgateway server up
```
$ jupyter kernelgateway \
    --port 9999 \
    --JupyterWebsocketPersonality.list_kernels=True
```
2. on another terminal run the jupyter lab up 
```
$ jupyter lab \
    --gateway-url=http://127.0.0.1:9999 \
    --ServerApp.session_manager_class=jupyter_server_synchronizer.SynchronizerSessionManager \
    --SynchronizerSessionManager.database_filepath=jupyter-database.db \
    --SynchronizerSessionManager.autosync=True \
    --SynchronizerSessionManager.log_level=DEBUG
```
3. Then observed the error msg been pop out on client side
```
.venv/lib/python3.8/site-packages/jupyter_server_synchronizer/manager.py", line 234, in list_sessions
        out = await super().list_sessions(self)
    TypeError: list_sessions() takes 1 positional argument but 2 were given
```

## Expected behavior

The expected behavior is to see it run without exception, not sure if it's some deps incompatible issue, but according to https://github.com/jupyter-server/synchronizer/blob/main/pyproject.toml, only seen to required `jupyter_server>=1.16.0` and `python >=3.7` 

## Context

- Operating System and version: Linux ubuntu 20.04
- Browser and version: Chrome
- Python: 3.8
- Jupyter Server version: 1.18.1
- Jupyter Lab version: 3.4.5
- Jupyter Server Synchronizer version: 0.0.8
- Jupyter Kernel Gateway version: 2.5.1

<details><summary>Command Line Output</summary>
<pre>
$ jupyter lab \
    --gateway-url=http://127.0.0.1:9999 \
    --ServerApp.session_manager_class=jupyter_server_synchronizer.SynchronizerSessionManager \
    --SynchronizerSessionManager.database_filepath=jupyter-database.db \
    --SynchronizerSessionManager.autosync=True \
    --SynchronizerSessionManager.log_level=DEBUG \
    --ip 0.0.0.0 \
    port 8080
[I 2022-08-15 22:20:32.616 ServerApp] jupyter_server_synchronizer | extension was successfully linked.
[I 2022-08-15 22:20:32.626 ServerApp] jupyterlab | extension was successfully linked.
[I 2022-08-15 22:20:32.634 ServerApp] nbclassic | extension was successfully linked.
[I 2022-08-15 22:20:32.784 ServerApp] notebook_shim | extension was successfully linked.
[W 2022-08-15 22:20:32.788 ServerApp] Config option `log_level` not recognized by `SynchronizerSessionManager`.
[I 2022-08-15 22:20:32.813 ServerApp] notebook_shim | extension was successfully loaded.
[W 2022-08-15 22:20:32.814 ServerApp] jupyter_server_synchronizer | extension failed loading with message: 'NoneType' object is not callable
[I 2022-08-15 22:20:32.815 LabApp] JupyterLab extension loaded from .venv/lib/python3.8/site-packages/jupyterlab
[I 2022-08-15 22:20:32.815 LabApp] JupyterLab application directory is .venv/share/jupyter/lab
[I 2022-08-15 22:20:32.818 ServerApp] jupyterlab | extension was successfully loaded.
[I 2022-08-15 22:20:32.821 ServerApp] nbclassic | extension was successfully loaded.
[I 2022-08-15 22:20:32.821 ServerApp] Serving notebooks from local directory: 
[I 2022-08-15 22:20:32.821 ServerApp] Jupyter Server 1.18.1 is running at:
[I 2022-08-15 22:20:32.821 ServerApp] Kernels will be managed by the Gateway server running at:
[I 2022-08-15 22:20:32.821 ServerApp] http://127.0.0.1:9999
[I 2022-08-15 22:20:32.821 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[W 2022-08-15 22:20:32.825 ServerApp] No web browser found: could not locate runnable browser.
[C 2022-08-15 22:20:32.825 ServerApp]

[E 2022-08-15 22:20:38.509 ServerApp] Uncaught exception GET /api/sessions?1660602038126 (172.16.5.55)
    HTTPServerRequest(protocol='http', host='localhost:8080', method='GET', uri='/api/sessions?1660602038126', version='HTTP/1.1', remote_ip='172.16.5.55')
    Traceback (most recent call last):
      File "venv/lib/python3.8/site-packages/tornado/web.py", line 1713, in _execute
        result = await result
      File ".venv/lib/python3.8/site-packages/jupyter_server/services/sessions/handlers.py", line 36, in get
        sessions = await ensure_async(sm.list_sessions())
      File ".venv/lib/python3.8/site-packages/jupyter_server/utils.py", line 182, in ensure_async
        result = await obj
      File ".venv/lib/python3.8/site-packages/jupyter_server_synchronizer/manager.py", line 234, in list_sessions
        out = await super().list_sessions(self)
    TypeError: list_sessions() takes 1 positional argument but 2 were given
</pre>
</details>

**Comment by kevin-bates** [comment]
Hi @chiawchen - thanks for opening this issue.  Looks like the `list_sessions()` call inadvertently included `self` over the course of the changes for #22 and #26.  Is this something you'd like to make a quick PR for?

**Comment by kevin-bates** [comment]
@Zsailer - FYI - I directed @chiawchen to this repo because they're interested in a form of "client-side HA" where they may want to update the client packages (bring down lab, refresh, restart lab, and reconnect to the persisted sessions that will be serviced by a gateway server).  I couldn't recall if this particular scenario can be detected and recovered from with the synchronizer - but I suspect it can.

**Comment by chiawchen** [comment]
sure thing! just add a small fix on it

**Comment by kevin-bates** [comment]
Addressed via #42

**Comment by Zsailer** [comment]
@chiawchen and @kevin-bates, great stuff here! Thank you for reporting and fixing this bug so quickly! 

I'll make a new release now.
